pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -932,6 +952,8 @@ async def _responses_create(
                 truncation=model_settings.get('openai_truncation', NOT_GIVEN),
                 timeout=model_settings.get('timeout', NOT_GIVEN),
                 service_tier=model_settings.get('openai_service_tier', NOT_GIVEN),
+                previous_response_id=previous_response_id
+                or model_settings.get('openai_previous_response_id', NOT_GIVEN),
```

Comment:
If a `openai_previous_response_id` was explicitly set, I think we should not do the stuff we do above, and assume the user only passed in the messages in message history that they actually meant to send